pyiceberg/catalog/dynamodb.py

```diff
@@ -28,6 +28,7 @@
 )
 
 import boto3
+from mypy_boto3_dynamodb.client import DynamoDBClient
```

Comment:
I think we should move this import below under the `TYPE_CHECKING` flag

Reply:
Good catch! I made a similar change in `glue.py` and also made the dev dependencies for these imports not optional under the dev group

Reply:
looks like I broke something!